staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -141,13 +141,21 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope RequestSco
 
 		status := http.StatusOK
 		// Return http.StatusAccepted if the resource was not deleted immediately and
-		// user requested cascading deletion by setting OrphanDependents=false.
+		// user requested cascading deletion by setting OrphanDependents=false, or
+		// PropagationPolicy=Foreground/Background.
 		// Note: We want to do this always if resource was not deleted immediately, but
 		// that will break existing clients.
 		// Other cases where resource is not instantly deleted are: namespace deletion
 		// and pod graceful deletion.
-		if !wasDeleted && options.OrphanDependents != nil && *options.OrphanDependents == false {
-			status = http.StatusAccepted
+		if !wasDeleted {
```

Comment:
hot take: If we're making a breaking change anyway, shouldn't we always return accepted if the object wasn't actually deleted, but will be in the future?

I'm not sure if it is acceptable to change this behavior.

@deads2k, @liggitt, or @caesarxuchao might remember why we settled on the existing code.